packages/shared/ReactFeatureFlags.js

```diff
@@ -24,6 +24,11 @@ export const enableNoopReconciler = false;
 // Experimental persistent mode (CS):
 export const enablePersistentReconciler = false;
 
+// Helps identify side effects in begin-phase lifecycle hooks:
+export const invokePrecommitLifecycleHooksTwice = false;
+// Helps identify side effects in setState callback:
+export const invokeSetStateCallbackTwice = false;
```

Comment:
Is there a scenario where it's useful for these to be separate flags?

Reply:
Not really. Maybe it was mostly just a naming concern, which is kind of silly I guess. Happy to combine them. Any suggestion for a name that covers both?

Reply:
`debugRenderPhaseSideEffects` seems appropriately generic :D

Reply:
![genius](https://user-images.githubusercontent.com/29597/32962428-401820ba-cb81-11e7-9b9a-32d4224ac107.gif)